biz.aQute.remote/src/aQute/remote/util/AgentSupervisor.java

```diff
@@ -41,16 +42,28 @@
 	}
 
 	protected void connect(Class<Agent> agent, Supervisor supervisor, String host, int port) throws Exception {
-		while (true)
+		connect(agent, supervisor, host, port, -1);
+	}
+
+	protected void connect(Class<Agent> agent, Supervisor supervisor, String host, int port, int timeout)
```

Comment:
@bjhargrave if a client calls `connect(agent, supervisor, host, port, -100)` it will hang indefinitely.  Is this OK?